apps/emqx_rule_engine/src/emqx_rule_engine.erl

```diff
@@ -259,7 +260,7 @@ create_resource(#{type := Type, config := Config0} = Params, Retry) ->
                 with_retry ->
                     %% Note that we will return OK in case of resource creation failure,
                     %% A timer is started to re-start the resource later.
-                    _ = (catch (?CLUSTER_CALL(init_resource, [M, F, ResId, Config]))),
+                    _ = (catch (?CLUSTER_CALL(init_resource_with_retrier, [M, F, ResId, Config]))),
```

Comment:
this call will crash during rolling upgrade.

Reply:
Fixed by falling back to `init_resource/4` if undefined.